### ISSUE TITLE
Per-user setting for ui mode (classic/notebookbar)

### ISSUE THREAD

**Issue by juliusknorr** [nextcloud/richdocuments#1975]
- Add an option to switch the UI mode per user
- PostMessage for switching ui_defaults on the fly should be also available -> https://github.com/CollaboraOnline/online/issues/3955

**Comment by juliushaertl** [comment]
Work in progress for adding a user interface switch into online: https://github.com/CollaboraOnline/online/pull/4254

Once that is done and the post messages are properly emitted, we can store the user preference and don't need an extra setting within Nextcloud.

**Comment by juliushaertl** [comment]
Steps left
- [ ] Catch post message and store it in the user preferences
- [ ] Add option to the user settings
- [ ] Add option to configure already existing app config value from the admin settings
- [ ] Maybe show a nice switcher with some UI screenshot on how the different classic/tabbed layouts look

**Comment by gitsan13** [comment]
Hi
Are you accepting a PR for this enhancement?
I would like to give it a try:)

**Comment by juliushaertl** [comment]
Sure, feel free to go ahead. Let us know if you have any questions :)

**Comment by juliusknorr** [comment]
After https://github.com/CollaboraOnline/online/pull/4254 and discussing with @jancborchardt we came to the conclusion that this is enough to have this within Collabora.

**Comment by jancborchardt** [comment]
Yup, nicely discoverable and it's cool that the picked view mode is just remembered. Way simpler than a setting. :)
